### PR TITLE
plots: Replace hardcoded TEMPLATES_CHOICES

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvc/issues/7944#issuecomment-1174713966

### CODE REVIEW

**Comment by daavoo** [comment]
> I believe that dynamic template choices will result in loading everything from `dvc/repo/plots` which in result will heavily impact on all of the commands. Thats why the benchmarks are failing.

Looked like a random failure. I re-triggered and benchmark passed

**Comment by pared** [comment]
Huh, interesting, I guess extracting rendering to `dvc_render` resulted also in moving the templates out of `dvc/repo/plots`.

**Review by pared — APPROVED** [review]
Could you remove the:
`tests/unit/command/test_plots.py::test_plots_templates_choices`?

**Comment by daavoo** [comment]
@pared Looks like now benchmarks are consistently failing. The mean increase is ~10ms so not sure how relevant it is

**Comment by dberenbaum** [comment]
> @pared Looks like now benchmarks are consistently failing. The mean increase is ~10ms so not sure how relevant it is

I think it's fine.

**Comment by skshetry** [comment]
> @pared Looks like now benchmarks are consistently failing. The mean increase is ~10ms so not sure how relevant it is

Please open an issue for this on `dvc-render` and try to fix it please. It might look small, but in the future, `dvc-render` might import something big and we'll regress.

Also, do we really need this PR? `dvc-render` can support different templates, but dvc could specify what it wants.

File: dvc/commands/plots.py [hunk 1]
```diff
@@ -222,14 +222,9 @@ def run(self):
...
-        "simple",
-        "linear",
-        "confusion",
-        "confusion_normalized",
-        "scatter",
-        "smooth",
-    ]
+    from dvc_render.vega_templates import TEMPLATES
+
+    TEMPLATES_CHOICES = [t.DEFAULT_NAME for t in TEMPLATES]
```
Comment (skshetry) [inline dvc/commands/plots.py:227]: Please move this to file level, there is no difference except this pollutes the class namespace.

**Comment by daavoo** [comment]
> > @pared Looks like now benchmarks are consistently failing. The mean increase is ~10ms so not sure how relevant it is
> 
> Please open an issue for this on `dvc-render` and try to fix it please. It might look small, but in the future, `dvc-render` might import something big and we'll regress.

It only imports std library and custom exception:

https://github.com/iterative/dvc-render/blob/4fe35d8b4c13d442d7ffa7a5853302dfab8de219/src/dvc_render/vega_templates.py#L1-L8

> Also, do we really need this PR? `dvc-render` can support different templates, but dvc could specify what it wants.

It doesn't make sense to have a hardcoded list. If a template is added to `dvc-render`, it is meant to be used by `dvc`.

**Comment by skshetry** [comment]
> > > @pared Looks like now benchmarks are consistently failing. The mean increase is ~10ms so not sure how relevant it is
> > 
> > 
> > Please open an issue for this on `dvc-render` and try to fix it please. It might look small, but in the future, `dvc-render` might import something big and we'll regress.
> 
> It only imports std library and custom exception:

It's not `.vega_templates` that I worry about, but the top level one where it imports from submodules: https://github.com/iterative/dvc-render/blob/4fe35d8b4c13d442d7ffa7a5853302dfab8de219/src/dvc_render/__init__.py#L4-L10

It seems `dvc-render` does not have heavy dependencies, so it looks good fine for now. But, please keep in mind to lazily load in `dvc-render` if you'll have any big imports in the future.

**Comment by pared** [comment]
So, this is why those hardcoded templates did exist - we wanted to separate cmd from plots. Now we would re-add smaller dependency. Maybe we should actually consider having a command to display the templates. Then we would avoid this issue.
I think this idea resurfaced during `templates` docs introduction, but I cannot find the particular comment.

**Comment by efiop** [comment]
What's the status here? CC @daavoo

**Comment by pared** [comment]
Addressed by #8129

**Pull request closed without merging**